src/main/java/com/streamr/client/StreamrClient.java

```diff
@@ -370,15 +366,12 @@ public void publish(Stream stream, Map<String, Object> payload, Date timestamp,
     }
 
     public void publish(Stream stream, Map<String, Object> payload, Date timestamp, String partitionKey, String groupKeyHex) {
-        if (!getState().equals(StreamrClient.State.Connected)) {
-            connect();
-        }
         if (groupKeyHex != null) {
             options.getEncryptionOptions().getPublisherGroupKeys().put(stream.getId(), groupKeyHex);
         }
         StreamMessage streamMessage = msgCreationUtil.createStreamMessage(stream, payload, timestamp, partitionKey, groupKeyHex);
         PublishRequest req = new PublishRequest(streamMessage, getSessionToken());
-        getWebsocket().send(req.toJson());
+        send(req);
```

Comment:
We should probably use same `send` method as the rest of the control messages. I wonder if there was an intentional reason to use `webSocket.send` directly here instead of our wrapped `send` method?